doc/source/roadmap-detailed.rst

```diff
@@ -0,0 +1,385 @@
+  still needs a CSC/CSR equivalent) and okay performance-wise.
+- Add support for ``pydata/sparse`` to ``scipy.sparse.linalg`` (and perhaps to
+  ``scipy.sparse.csgraph`` after that).
+- Indicate in the documentation that for new code users should prefer
+  ``pydata/sparse`` over sparse matrices.
+- When NumPy deprecates ``numpy.matrix``, vendor that or maintain it as a
+  stand-alone package.
+
+Regarding the different sparse matrix formats: there are a lot of them.  These
+should be kept, but improvements/optimizations should go into CSR/CSC, which
+are the preferred formats.  LIL may be the exception, it's inherently
+inefficient.  It could be dropped if DOK is extended to support all the
+operations LIL currently provides.
+
+
+sparse.csgraph
+``````````````
+This module is in good shape.
+
+
+sparse.linalg
+`````````````
+Arpack is in good shape.
+
+isolve:
+
+    - callback keyword is inconsistent
+    - tol keyword is broken, should be relative tol
+    - Fortran code not re-entrant (but we don't solve, maybe re-use from
+      PyKrilov)
+
+dsolve:
+
+    - add sparse Cholesky or incomplete Cholesky
+    - look at CHOLMOD
+
+
+Ideas for new features:
+
+    - Wrappers for PROPACK for faster sparse SVD computation.
+
+
+spatial
+```````
+QHull wrappers are in good shape, as is ``cKDTree``.
+
+Needed:
+
+    - ``KDTree`` will be removed, and ``cKDTree`` will be renamed to ``KDTree``
+      in a backwards-compatible way.
+    - ``distance_wrap.c`` needs to be cleaned up (maybe rewrite in Cython).
+
+
+special
+```````
+Though there are still a lot of functions that need improvements in precision,
+probably the only show-stoppers are hypergeometric functions, parabolic cylinder
+functions, and spheroidal wave functions. Three possible ways to handle this:
+
+    1. Get good double-precision implementations. This is doable for parabolic
+       cylinder functions (in progress). I think it's possible for hypergeometric
+       functions, though maybe not in time. For spheroidal wavefunctions this is
+       not possible with current theory.
+
+    2. Port Boost's arbitrary precision library and use it under the hood to get
+       double precision accuracy. This might be necessary as a stopgap measure
+       for hypergeometric functions; the idea of using arbitrary precision has
+       been suggested before by @nmayorov and in
+       `gh-5349 <https://github.com/scipy/scipy/issues/5349>`__.  Likely
+       necessary for spheroidal wave functions, this could be reused:
+       https://github.com/radelman/scattering.
+
+    3. Add clear warnings to the documentation about the limits of the existing
+       implementations.
+
+
+stats
+`````
+
+This module is in good shape overall.  New functionality that's similar to
+what's already present can continue to be added; more advanced statistical
+routines may fit better in ``statsmodels``.  Some ideas for new contributions
+are:
+
```

Comment:
Maybe also mention Bunches: almost PR-ready implementations are well hidden in (possibly closed )issues concerning linregress.

Reply:
good idea, added